app/models/support/service_performance/device_counts.rb

```diff
@@ -0,0 +1,70 @@
+class Support::ServicePerformance::DeviceCounts
+  OVER_ORDERED_ALLOCATION_QUERY = 'devices_ordered > 0 AND cap < devices_ordered'
+  NOT_OVER_ORDERED_ALLOCATION_QUERY = 'cap >= devices_ordered'
+
+  def self.sum_allocation(
+    school_scope:,
+    device_type:,
+    scope_query: nil,
+    who_will_order_devices:,
+    sum_expression:
+  )
+    SchoolDeviceAllocation
+      .where(device_type: device_type)
+      .where(scope_query)
+      .joins(school: :preorder_information)
+      .merge(school_scope)
+      .where(preorder_information: {
+        who_will_order_devices: who_will_order_devices
+      })
+      .sum(sum_expression)
+  end
+
+  attr_reader :key
+
+  def initialize(
+    key:,
+    school_scope:,
+    device_type:,
+    who_will_order_devices:
+  )
+    @key = key
+    @school_scope = school_scope
+    @device_type = device_type
+    @who_will_order_devices = who_will_order_devices
+  end
+
+  def available
+    sum_allocation(sum_expression: 'cap')
+  end
+
+  def ordered
+    sum_allocation(sum_expression: 'devices_ordered')
+  end
+
+  def remaining_excl_over_ordered
+    sum_allocation(
+      scope_query: NOT_OVER_ORDERED_ALLOCATION_QUERY,
+      sum_expression: 'cap - devices_ordered'
+    )
+  end
+
+  def over_ordered
+    -1 * sum_allocation(
```

Comment:
Could rewrite this as `-sum_allocation(`

Reply:
Yep, but – and it's more of a personal thing – I find this more readable (I find `-expression` _really_ easy to gloss over too quickly and miss that detail.